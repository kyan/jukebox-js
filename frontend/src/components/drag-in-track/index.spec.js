import React from 'react'
import { shallow } from 'enzyme'
import DragInTrack from './index'

describe('DragInTrack', () => {
  const onDropMock = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('render', () => {
    describe('when controls disabled', () => {
      it('renders as expected', () => {
        const wrapper = shallow(
          <DragInTrack disabled onDrop={onDropMock}>
            <span>Hello World</span>
          </DragInTrack>
        )

        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('when controls enabled', () => {
      it('renders as expected', () => {
        const wrapper = shallow(
          <DragInTrack disabled={false} onDrop={onDropMock}>
            <span>Hello World</span>
          </DragInTrack>
        )

        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
