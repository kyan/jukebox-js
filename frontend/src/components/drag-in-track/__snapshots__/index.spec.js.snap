// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragInTrack render when controls disabled renders as expected 1`] = `
<span>
  Hello World
</span>
`;

exports[`DragInTrack render when controls enabled renders as expected 1`] = `
<Component
  backend={[Function]}
>
  <DropTarget(UrlDropArea)
    accepts={
      Array [
        "__NATIVE_URL__",
      ]
    }
    onDrop={[MockFunction]}
  >
    <span>
      Hello World
    </span>
  </DropTarget(UrlDropArea)>
</Component>
`;
